rakelib/form526.rake

```diff
@@ -226,4 +226,127 @@ namespace :form526 do
     end
     puts "reuploaded files for #{form_submissions.count} submissions"
   end
+
+  desc 'Convert SIP data to camel case and fix checkboxes [/export/path.csv, ids]'
+  task :convert_sip_data, [:csv_path] => :environment do |_, args|
+    raise 'No CSV path provided' unless args[:csv_path]
+
+    ids = args.extras || []
+
+    def to_olivebranch_case(val)
+      OliveBranch::Transformations.transform(
+        val,
+        OliveBranch::Transformations.method(:camelize)
+      )
+    end
+
+    def un_camel_va_keys!(hash)
+      json = hash.to_json
+      # rubocop:disable Style/PerlBackrefs
+      # gsub with a block explicitly sets backrefs correctly https://ruby-doc.org/core-2.6.6/String.html#method-i-gsub
+      json.gsub!(OliveBranch::Middleware::VA_KEY_REGEX) do
+        key = $1
+        "#{key.gsub('VA', 'Va')}:"
+      end
+      JSON.parse(json)
+      # rubocop:enable Style/PerlBackrefs
+    end
+
+    # make a hash that has both the original and corrupted versions of the disability name.
+    # eg {"myocardial infarction (mi)" => "myocardial infarction (MI)"}
+    def get_disability_array(form_data_hash)
+      new_conditions = form_data_hash['newDisabilities']&.collect { |d| d.dig('condition') } || []
+      rated_disabilities = form_data_hash['ratedDisabilities']&.collect { |rd| rd['name'] } || []
+      new_conditions + rated_disabilities
+    end
+
+    # downcase and remove everything but letters and numbers
+    def simplify_string(string)
+      string&.downcase&.gsub(/[^a-z0-9]/, '')
+    end
+
+    def get_dis_translation_hash(disability_array)
+      dis_translation_hash = {}
+      disability_array.each do |dis|
+        dis_translation_hash[simplify_string(dis)] = dis
+      end
+      dis_translation_hash
+    end
+
+    def fix_treatment_facilities_disability_name(form_data_hash, dis_translation_hash, disability_array)
+      transformed = false
+      # fix vaTreatmentFacilities -> treatedDisabilityNames
+      # this should never happen, just want to confirm
+      form_data_hash['vaTreatmentFacilities']&.each do |va_treatment_facilities|
+        new_treated_disability_names = {}
+        if va_treatment_facilities['treatedDisabilityNames']
+          va_treatment_facilities['treatedDisabilityNames'].each do |disability_name, value|
+            if disability_array.include? disability_name
+              new_treated_disability_names[disability_name] = value
+            else
+              transformed = true
+              original_disability_name = dis_translation_hash[simplify_string(disability_name)]&.downcase
```

Comment:
Thanks for [pointing out](https://github.com/department-of-veterans-affairs/vets-api/pull/5651/commits/2d9275f2287d5e801f481536b49387ceaf064197) why we're doing this. I feel like it might be good to put a comment here --it would have definitely caught me off guard if you hadn't pointed it out.